src/ai.c

```diff
@@ -1874,6 +1876,111 @@ static int aiL_face( lua_State *L )
 
 
 /**
+ * @brief Gives the direction to follow in order to reach the target while
+ *  minimizating risk.
+ *
+ * This method is based on a simplified version of trajectory generation in
+ * mobile robotics using the potential method.
+ *
+ * The principle is to consider the mobile object (ship) as a mechanical object.
+ * Obstacles (enemies) and the target exert 
+ * attractive or repulsive force on this object.
+ *
+ * Only visible ships are taken into account.
+ *
+ * @usage ai.careful_face( a_pilot ) -- Face a pilot
+ * @usage ai.careful_face( a_vector ) -- Face a vector
+ *
+ *    @luaparam target Target to go to.
+ * @luafunc careful_face( target )
+ */
+static int aiL_careful_face( lua_State *L )
+{
+   LuaVector *lv;
+   Vector2d *tv, F, F1;
+   Pilot* p;
+   Pilot *p_i;
+   double k_diff, k_goal, k_enemy, k_mult,
+          d, diff, dist, factor;
+   int enemy, i;
+
+   /* Init some variables */
+   p = cur_pilot;
+
+   /* Get first parameter, aka what to face. */
+   if (lua_ispilot(L,1)) {
+      p = luaL_validpilot(L,1);
+      /* Target vector. */
+      tv = &p->solid->pos;
+   }
+   else if (lua_isnumber(L,1)) {
+      d = (double)lua_tonumber(L,1);
+      if (d < 0.)
+         tv = &cur_pilot->solid->pos;
+      else
+         NLUA_INVALID_PARAMETER(L);
+   }
+   else if (lua_isvector(L,1)) {
+      lv = lua_tovector(L,1);
+      tv = &lv->vec;
+   }
+   else
+      NLUA_INVALID_PARAMETER(L);
+
+   /* Default gains. */
+   k_diff = 10.;
+   k_goal = 1.;
+   k_enemy = 4000000.;
+
+   /* Init the force */
+   vect_cset( &F, 0., 0.) ;
+   vect_cset( &F1, tv->x - cur_pilot->solid->pos.x, tv->y - cur_pilot->solid->pos.y) ;
+   dist = VMOD(F1) + 0.1; /* Avoid / 0*/
+   vect_cset( &F1, F1.x * k_goal / dist, F1.y * k_goal / dist) ;
+
+   /*cycle through all the pilots in order to compute the force */
+   for(i = 0; i<pilot_nstack; i++)
+   {
+
+      p_i = pilot_stack[i];
+
+      /* Valid pilot isn't self, is in range, isn't the target and isn't disabled */
+      if(p_i->id != cur_pilot->id && pilot_inRangePilot( cur_pilot, p_i) == 1  
+            && p_i->id != p->id && !pilot_isDisabled(p_i) )
+
+      {
+           dist = vect_dist(&p_i->solid->pos, &cur_pilot->solid->pos) + 0.1; /* Avoid / 0*/
+           k_mult = pilot_relhp( p_i, cur_pilot );
+
+           /* Check if friendly or not */
+           if (p_i->faction == FACTION_PLAYER)
```

Comment:
Is there any specific reason you check for player instead of letting areEnemies handle it? (besides small speed boost it may give)